File: frontend/src/lib/api/export.ts
import { apiFetch } from "@/lib/api/client";
import type { Job } from "@/lib/api/jobs";

export interface ExportOptions {
  width?: number;
  height?: number;
  fps?: number;
  crf?: number;
  fade_duration?: number;
}

export function startExport(projectId: string, options: ExportOptions = {}) {
  return apiFetch<Job>(`/projects/${projectId}/export`, {
    method: "POST",
    json: {
      width: options.width ?? 1280,
      height: options.height ?? 720,
      fps: options.fps ?? 24,
      crf: options.crf ?? 23,
      fade_duration: options.fade_duration ?? 0.5,
    },
  });
}

File: backend/app/schemas/export.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ExportRequest(BaseModel):
    width: int = Field(default=1280, ge=128, le=3840)
    height: int = Field(default=720, ge=72, le=2160)
    fps: int = Field(default=24, ge=1, le=60)
    crf: int = Field(default=23, ge=0, le=51)
    fade_duration: float = Field(default=0.5, ge=0.0, le=3.0)
